SoomlaiOSProfile/SoomlaProfile.m

```diff
@@ -100,9 +104,14 @@ - (void)loginWithProvider:(Provider)provider andPayload:(NSString *)payload andR
         [authController loginWithProvider:provider andAutoLogin:NO andPayload:payload andReward:reward];
     }
     @catch (NSException *exception) {
-
-        // TODO: implement logic like in java that will raise the exception. Currently not raised
-        [socialController loginWithProvider:provider andAutoLogin:NO andPayload:payload andReward:reward];
+        @try {
+            // TODO: implement logic like in java that will raise the exception. Currently not raised
+            [socialController loginWithProvider:provider andAutoLogin:NO andPayload:payload andReward:reward];
+        }
+        @catch (NSException *exception) {
+            // TODO: implement logic like in java that will raise the exception. Currently not raised
+            [gameServicesController loginWithProvider:provider andAutoLogin:NO andPayload:payload andReward:reward];
```

Comment:
@EugeneButusov, can we avoid such bad things? Maybe in a separate task. I really dislike this.